sdk/src/main/scala/io/horizen/account/state/AccountStateView.scala

```diff
@@ -86,32 +87,76 @@ class AccountStateView(
 
   // after this we always reset the counters
   override def getFeePaymentsInfo(
-      withdrawalEpoch: Int,
-      consensusEpochNumber: ConsensusEpochNumber,
-      distributionCap: BigInteger,
-      blockToAppendFeeInfo: Option[AccountBlockFeeInfo] = None
+    withdrawalEpoch: Int,
+    consensusEpochNumber: ConsensusEpochNumber,
+    distributionCap: BigInteger,
+    blockToAppendFeeInfo: Option[AccountBlockFeeInfo] = None,
   ): (Seq[AccountPayment], BigInteger) = {
     var blockFeeInfoSeq = metadataStorageView.getFeePayments(withdrawalEpoch)
     blockToAppendFeeInfo.foreach(blockFeeInfo => blockFeeInfoSeq = blockFeeInfoSeq :+ blockFeeInfo)
     val mcForgerPoolRewards = getMcForgerPoolRewards(consensusEpochNumber, distributionCap)
     val poolBalanceDistributed = mcForgerPoolRewards.values.foldLeft(BigInteger.ZERO)((a, b) => a.add(b))
     metadataStorageView.updateMcForgerPoolRewards(mcForgerPoolRewards)
-    (AccountFeePaymentsUtils.getForgersRewards(blockFeeInfoSeq, mcForgerPoolRewards), poolBalanceDistributed)
+    if (Version1_4_0Fork.get(consensusEpochNumber).active) {
```

Comment:
This method overrides a `BaseStateReader` one, but actually it is not a read-only because updates the `metadataStorageView`. Not really an issue but it would be nice not to break the semantic of the base reader class. Moreover, the same method overridden by `AccountState` does not use  the last two parameters.

Reply:
> This method overrides a `BaseStateReader` one, but actually it is not a read-only because updates the `metadataStorageView`. Not really an issue but it would be nice not to break the semantic of the base reader class. Moreover, the same method overridden by `AccountState` does not use the last two parameters.

I would not consider the read-only break a big issue, was already introduced in past version